dotcom-rendering/src/web/components/SignInGate/types.ts

```diff
@@ -18,15 +18,62 @@ export type SignInGateComponent = {
 	}: CanShowGateProps) => Promise<boolean>;
 };
 
+// Taking this approach for validating whether a specific
+// string is in a union type as per the advice of this blog:
+// https://dev.to/hansott/how-to-check-if-string-is-member-of-union-type-1j4m
+// Other option is to use an enum, though there are downside to this:
+// https://dev.to/azure/the-dangers-of-typescript-enums-55pd
+
+export const ALL_USER_TYPES = ['new', 'guest', 'current'] as const;
```

Comment:
Using a union type here, but could as well use an enum. Are there any particular style guide rules here?

Reply:
Disccussed with Olly, and looks ok